pubsub/dataprovider.go

```diff
@@ -47,9 +49,9 @@ func (p *PriceDispatcher) process(ctx context.Context) {
 			log.Println("Price producer is done")
 			return
 		case price := <-p.priceProducer.Out():
-			if _, ok := p.subscriptions[price.Symbol]; ok {
+			if ch, ok := p.subscriptions[price.Symbol]; ok {
 				log.Println("Found subscription ", price.Symbol, " for price ", price.Price)
-				p.subscriptions[price.Symbol] <- price
+				ch <- price
```

Comment:
I just realized it is entirely wrong.
If there are multiple subscriptions to the same channel, then only one of the subscriber will receive the message, because you're not broadcasting to all subscribers just randomly sending it to one.